### PR TITLE
Use flags for redis settings

### PR DESCRIPTION
**Pull request by tempoz**
Additionally, we use register functions to move the implementation details of redis out of
the main functions.

---

**Version bump**: Minor

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1771** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1771" target="_blank">Graphite</a> 
    * **PR #1772** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1772" target="_blank">Graphite</a> 
      * **PR #1773** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1773" target="_blank">Graphite</a> 
        * **PR #1774** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774" target="_blank">Graphite</a> 
          * **PR #1775** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1775" target="_blank">Graphite</a> 
            * **PR #1776** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1776" target="_blank">Graphite</a> 
              * **PR #1777** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1777" target="_blank">Graphite</a> 
                * **PR #1778** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778" target="_blank">Graphite</a> 
                  * **PR #1779** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1779" target="_blank">Graphite</a> 
                    * **PR #1780** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1780" target="_blank">Graphite</a> 
                      * **PR #1783** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1783" target="_blank">Graphite</a> 
                      * **PR #1833** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833" target="_blank">Graphite</a>  (this PR)
                        * **PR #1834** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1834" target="_blank">Graphite</a> 
                          * **PR #1835** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1835" target="_blank">Graphite</a> 
                            * **PR #1836** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1836" target="_blank">Graphite</a> 
                              * **PR #1837** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1837" target="_blank">Graphite</a> 
                                * **PR #1838** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1838" target="_blank">Graphite</a> 
                                  * **PR #1839** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1839" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).

File: enterprise/server/util/redisutil/redisutil.go
Comment (bduffany) [inline enterprise/server/util/redisutil/redisutil.go:31]: I made this comment in another PR too, but wdyt about putting this in a separate package like `enterprise/server/util/redisconf` or similar? It would be nice to keep this package as a collection of generic redis utils, rather than having any BB-specific stuff

Comment (tempoz) [reply enterprise/server/util/redisutil/redisutil.go]: I moved all this into `enterprise/server/backends/redis_client/redis_client.go`, per Tyler's suggestion. Same with the similar things in the cache aand remote_execution PRs ^-^

Thanks for letting me know about this, I didn't know about this goal of `util` before!

File: enterprise/server/backends/redis_client/redis_client.go
Comment (bduffany) [inline enterprise/server/backends/redis_client/redis_client.go:54]: nit: consider making this read at the callsite like `redis_client.RegisterDefault()` or `redis_client.RegisterDefaultClient()`

File: enterprise/server/backends/redis_client/redis_client.go
Comment (bduffany) [inline enterprise/server/backends/redis_client/redis_client.go:53]: nit: looks like this TODO is already done?

File: enterprise/server/backends/redis_client/redis_client.go
Comment (bduffany) [inline enterprise/server/backends/redis_client/redis_client.go:39]: uber nit: DefaultRedisClientConfig isn't a common acronym (yet?) so IMO something like `cfg`, `conf`, or `config` would be a more easily grokkable abbreviation (same below)

File: enterprise/server/backends/redis_client/redis_client.go
Comment (bduffany) [inline enterprise/server/backends/redis_client/redis_client.go:21]: nit: make private

**Review by bduffany — APPROVED** [review]
LGTM, just nits

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: UWH9zAeATn6HjbQfFiuS

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: LYGXjCZ9jzXE0ra5XapW

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: bO0bzB7R9g7bK8rQ9uPT

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: x9vSQMZuwVwdzOiJU1Ph

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: xLhUv52COpx6k7AsRAsn

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1833?utm_source=stack-comment).


Job Reference: KZCXEHPsl7FVcdrah9Au

**Pull request merged**
